### ISSUE TITLE
git submodule update not working on macOS

### ISSUE THREAD

**Issue by stephanschulz** [zkmkarlsruhe/ofxTensorFlow2#35]
I get the following access rights issue when following the Quick Start guide. Do I need to fork the add-on first?

```
stephanschulz@Stephans-Komputer ofxTensorFlow2 % git submodule update --init --recursive
Cloning into '/Applications/of_v20220530_osx_release/addons/ofxTensorFlow2/libs/cppflow'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:serizba/cppflow.git' into submodule path '/Applications/of_v20220530_osx_release/addons/ofxTensorFlow2/libs/cppflow' failed
Failed to clone 'libs/cppflow'. Retry scheduled
Cloning into '/Applications/of_v20220530_osx_release/addons/ofxTensorFlow2/libs/cppflow'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:serizba/cppflow.git' into submodule path '/Applications/of_v20220530_osx_release/addons/ofxTensorFlow2/libs/cppflow' failed
Failed to clone 'libs/cppflow' a second time, aborting
```

**Comment by danomatika** [comment]
We don’t have control of that. It could just have been a hiccup on the github servers and when it happens, I usually wait and try again later.

enohp ym morf tnes
-----------
Dan Wilcox
danomatika.com
robotcowboy.com

**Comment by danomatika** [comment]
Is this resolved by now?

**Comment by stephanschulz** [comment]
I just tried it again and still gives me the same error. 
But downloading the cpp directly and placing is in the cpp folder works. 
![Screen Shot 2023-02-27 at 2 19 52 PM](https://user-images.githubusercontent.com/1054816/221661723-cc27503c-c520-49a9-87e7-36315eae9b09.jpg)

**Comment by danomatika** [comment]
Can you try the following?

1. edit .gitmodules and change: `git@github.com:serizba/cppflow.git` -> `https://github.com/serizba/cppflow.git`
2. run
~~~
git submodule deinit -f libs/cppflow
git submodule update --init --recursive
git submodule sync
~~~

**Comment by stephanschulz** [comment]
thanks for continuing to pay attention to this :)

I edited the .gitmodules.
I do not sure what to do with "2. run"

calling `git submodule deinit -f libs/cppflow` produces an error. see screenshot

![Screen Shot 2023-03-01 at 3 51 50 PM](https://user-images.githubusercontent.com/1054816/222262438-e7d36437-c48a-4d9b-a7e7-8e6f3fbd3b95.jpg)

**Comment by danomatika** [comment]
"git run" is not a command. The "2. run" in my previous post is an instruction to run the code which is quoted underneath in the Terminal.

Looking at your Terminal output, you indeed ran `git deinit` successfully once. The euro comes for running it again when the submodule has already been deinitialized. To continue, run the next two commands:
~~~
git submodule update --init --recursive
git submodule sync
~~~

**Comment by danomatika** [comment]
Also, if you manually copied the files in, you may need to manually remove the folder before running update.

**Comment by danomatika** [comment]
I think I know the issue. I bet the system you are cloning on does not have SSH keys which are copied to your Github account. The submodule is using git@github.com so it will try to use SSH for authentication. In any case, I will update the submodule url to https and make a 1.4.1 release.

**Comment by stephanschulz** [comment]
I will try the rest of the commands in a few days when I am back at home.

**Comment by stephanschulz** [comment]
yes this works. 
thank you

**Comment by danomatika** [comment]
Thanks for the confirmation. Now fixed in main and 1.4.1 tag.

**Comment by danomatika** [comment]
Thanks for confirming.
